src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -385,6 +385,12 @@ private void AssertNoLegacyReplicationViolation(ReplicationBatchItem item)
 
         private bool CanContinueBatch(ReplicationState state, ref long next)
         {
+            if (state.NumberOfItemsSent == 0)
+            {
+                // always send at least one item
+                return true;
```

Comment:
how this will affect a case when delay is set?

Reply:
fixed, placed this after `state.Delay` check